pygmt/tests/test_figure.py

```diff
@@ -52,6 +58,23 @@ def test_figure_region_country_codes():
     npt.assert_allclose(fig.region, np.array([0.0, 360.0, -90.0, 90.0]))
 
 
+def test_figure_repr():
+    """
+    Make sure that figure output's PNG and HTML printable representations look
+    ok.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 1, 2, 3], frame=True)
+    # Check that correct PNG 8-byte file header is produced
+    # https://en.wikipedia.org/wiki/Portable_Network_Graphics#File_header
+    repr_png = fig._repr_png_()  # pylint: disable=protected-access
```

Comment:
Is `_repr_png_` and `_repr_html_` useful? I have no idea how these functions can be used.

Reply:
Good question. They are used by [`IPython.display.display_png`](https://ipython.readthedocs.io/en/7.27.0/api/generated/IPython.display.html#IPython.display.display_png) and [`IPython.display.display_html`](https://ipython.readthedocs.io/en/7.27.0/api/generated/IPython.display.html#IPython.display.display_html) respectively, i.e. for display in Jupyter notebooks.

Technically I should have written these unit test using `IPython.display.display_png(fig)` and `IPython.display.display_html(fig)` instead of testing these non-public functions directly, but it's hard to test a pop-up figure. So I've settled for testing the `__repr_png__` and `__repr_html__` instead (which works with or without IPython installed). Con with this approach is that I need to do `pylint: disable=protected-access` :upside_down_face:

Reply:
> They are used by [`IPython.display.display_png`](https://ipython.readthedocs.io/en/7.27.0/api/generated/IPython.display.html#IPython.display.display_png) and [`IPython.display.display_html`](https://ipython.readthedocs.io/en/7.27.0/api/generated/IPython.display.html#IPython.display.display_html) respectively, i.e. for display in Jupyter notebooks.

PyGMT already provides the `Figure.show()` function for display in Jupyter notebooks. Removing `_repr_png_()` and `_repr_html_()` doesn't affect the `Figure.show()` function. Do you think we should remove these two functions?

Reply:
It might not affect `Figure.show()`, but I think we need to keep it. Having `__repr_png__()` and `__repr_html__` means that putting just `fig` at the last line of a Jupyter notebook cell will display the plot. E.g. like this:

![image](https://user-images.githubusercontent.com/23487320/133946061-a457f61f-b420-49b2-a045-8731bd97f72e.png)

This works in Jupyter/IPython because of a [`IPython.display.display_html(fig)`](https://ipython.readthedocs.io/en/7.27.0/api/generated/IPython.display.html#IPython.display.display_html) call. Similarly, if you put a `df` (pandas.DataFrame) on the last line of a cell, a nicely formatted html table will be printed because Jupyter/IPython does `IPython.display.display_html(df)` (see the `__repr_html__` at https://github.com/pandas-dev/pandas/blob/73c68257545b5f8530b7044f56647bd2db92e2ba/pandas/core/frame.py#L1007-L1049).